### PR TITLE
Add 'build at' time to webpack output

### PR DESCRIPTION
**Pull request by Hoten**
**What kind of change does this PR introduce?**

Feature

**Did you add tests for your changes?**

Yes

**If relevant, link to documentation update:**

N/A

**Summary**

Finishing work as started in this PR by @yeskunall and @selfawaresoup: https://github.com/webpack/webpack/pull/5026/

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: test/StatsTestCases.test.js [hunk 1]
```diff
@@ -73,14 +73,20 @@ describe("StatsTestCases", () => {
 				if(!hasColorSetting) {
 					actual = actual
 						.replace(/\u001b\[[0-9;]*m/g, "")
-						.replace(/[0-9]+(\s?ms)/g, "X$1");
+						.replace(/[0-9]+(\s?ms)/g, "X$1")
+						.replace(
+							/[A-z]{3} [A-z]{3} \d\d \d{4} \d\d:\d\d:\d\d GMT[+-]\d{4}( \(\w{3,4}\))?/g,
```
Comment (Hoten) [inline test/StatsTestCases.test.js]: looks like this regex isn't quite right - see errors on appveyor: https://ci.appveyor.com/project/sokra/webpack/build/1.0.6407/job/7n2u59afbs3gah5a

Comment (Hoten) [reply test/StatsTestCases.test.js]: gonna change "\w{3,4}" to "\w+"

**Comment by Hoten** [comment]
Not having much luck diagnosing the error here.

**Comment by ooflorent** [comment]
Could you rebase your PR on next branch?

**Comment by Hoten** [comment]
Ran through the whole "git rebase next" process. Never rebased something before, possible I messed it up...

Do I need to open a new PR?

**Comment by ooflorent** [comment]
No. You just need the change the target branch of the PR to next also.

**Comment by ooflorent** [comment]
Why are dates included twice??

**Comment by Hoten** [comment]
Bad merge! Just removed that duplication.

**Comment by Hoten** [comment]
Sorry for all the noise here. Should I squash some of these commits?

File: lib/Stats.js [hunk 2]
```diff
@@ -616,10 +617,18 @@ class Stats {
...
+			colors.normal("Built at: ");
+			colors.normal(`${now[0]} ${now[1]} ${now[2]} ${now[3]} `);
+			colors.bold(now[4]);
+			colors.normal(now[6] ? ` ${now[5]} ${now[6]}` : ` ${now[5]}`);
+			newline();
+		}
 		if(obj.env) {
 			colors.normal("Environment (--env): ");
 			colors.bold(JSON.stringify(obj.env, null, 2));
-			newline();
```
Comment (ooflorent) [inline lib/Stats.js]: Keep the semi please

File: lib/Stats.js [hunk 3]
```diff
@@ -252,6 +252,7 @@ class Stats {
 		if(showHash) obj.hash = this.hash;
 		if(showTimings && this.startTime && this.endTime) {
 			obj.time = this.endTime - this.startTime;
+			obj.builtAt = this.endTime;
```
Comment (sokra) [inline lib/Stats.js]: Don't put it under `showTimings` create a separate option.

File: lib/Stats.js [hunk 4]
```diff
@@ -616,6 +617,14 @@ class Stats {
 			colors.normal("ms");
 			newline();
 		}
+		if(typeof obj.builtAt === "number") {
+			const now = (new Date(obj.builtAt)).toString().split(" ");
+			colors.normal("Built at: ");
+			colors.normal(`${now[0]} ${now[1]} ${now[2]} ${now[3]} `);
```
Comment (sokra) [inline lib/Stats.js]: Date.toLocaleDateString()

File: lib/Stats.js [hunk 5]
```diff
@@ -616,6 +617,14 @@ class Stats {
...
+			colors.bold(now[4]);
```
Comment (sokra) [inline lib/Stats.js]: Date.toLocaleTimeString()

**Comment by sokra** [comment]
Thanks

**Pull request merged**